test/features/upsert/upsert_test.go

```diff
@@ -85,6 +91,66 @@ func TestUpsert(t *testing.T) {
 			massert.Equal(t, expected, actual)
 		},
 	}, {
+		name: "CreateOrUpdate when record does not exist",
+		run: func(t *testing.T, client *PrismaClient, ctx cx) {
+			actual, err := client.Post.UpsertOne(
+				Post.ID.Equals("upsert"),
+			).CreateOrUpdate(
+				Post.Title.Set("title"),
+				Post.Views.Set(0),
+			).Exec(ctx)
+			if err != nil {
+				t.Fatalf("fail %s", err)
+			}
+
+			expected := &PostModel{
+				InnerPost: InnerPost{
+					ID:    actual.ID,
+					Title: "title",
+					Views: 0,
+				},
+			}
+
+			massert.Equal(t, expected, actual)
+		},
+	}, {
+		name: "CreateOrUpdate when record exists",
+		before: []string{`
+		mutation {
+			result: createOnePost(data: {
+				id: "upsert",
+				title: "title",
+				views: 0,
+			}) {
+				id
+			}
+		}
+	`},
```

Comment:
The graphql protocol is unfortunately already deprecated, so let's just write this in the test as regular Go code.